internal/bpf/bpfWrapper.c

```diff
@@ -190,18 +208,18 @@ int Load_attach_bpf_xdp_pass(char *ifname) {
 		 __FUNCTION__, ifname, ifindex);
 
 	/* Load the BPF program */
-	err = bpf_prog_load(filename, BPF_PROG_TYPE_XDP, &obj, &prog_fd);
-	if (err < 0) {
-		Log_Error("%s: Couldn't load BPF-OBJ file(%s)\n", __FUNCTION__, filename);
-		return -1;
-	}
+	 err = bpf_xdp_query_id(ifindex, (int) xdp_flags, &prog_fd);

```

Comment:
we might need to look at this from a libxdp perspective. I think bpf_xdp_query_id might be coming from libbpf.
@maryamtahhan might know if there's a libxdp equivalent?
we can look into it anyway